pkg/blockbuilder/scheduler/config.go

```diff
@@ -11,18 +11,24 @@ import (
 )
 
 type Config struct {
-	BuilderConsumerGroup   string        `yaml:"builder_consumer_group"`
-	SchedulerConsumerGroup string        `yaml:"scheduler_consumer_group"`
-	SchedulingInterval     time.Duration `yaml:"kafka_monitor_interval"`
+	BuilderConsumerGroup string        `yaml:"builder_consumer_group"`
+	SchedulingInterval   time.Duration `yaml:"kafka_monitor_interval"`
+	ConsumeInterval      time.Duration `yaml:"consume_interval"`
+	StartupObserveTime   time.Duration `yaml:"startup_observe_time"`
+	JobLeaseTime         time.Duration `yaml:"job_lease_time"`
+	UpdateJobInterval    time.Duration `yaml:"update_job_interval"`
 
 	// Config parameters defined outside the block-builder-scheduler config and are injected dynamically.
 	Kafka ingest.KafkaConfig `yaml:"-"`
 }
 
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
-	f.StringVar(&cfg.BuilderConsumerGroup, "block-builder-scheduler.builder-consumer-group", "block-builder", "The Kafka consumer group used by block-builders.")
-	f.StringVar(&cfg.SchedulerConsumerGroup, "block-builder-scheduler.scheduler-consumer-group", "block-builder-scheduler", "The Kafka consumer group used by block-builder-scheduler.")
+	f.StringVar(&cfg.BuilderConsumerGroup, "block-builder-scheduler.builder-consumer-group", "block-builder", "The Kafka consumer group used for getting/setting commmitted offsets.")
 	f.DurationVar(&cfg.SchedulingInterval, "block-builder-scheduler.scheduling-interval", 20*time.Second, "How frequently to recompute the schedule.")
+	f.DurationVar(&cfg.ConsumeInterval, "block-builder-scheduler.consume-interval", 1*time.Hour, "Interval between consumption cycles.")
+	f.DurationVar(&cfg.StartupObserveTime, "block-builder-scheduler.startup-observe-time", 25*time.Second, "How long to observe worker state before scheduling jobs.")
+	f.DurationVar(&cfg.JobLeaseTime, "block-builder-scheduler.job-lease-time", 2*time.Minute, "How long a job lease will live for before expiring.")
```

Comment:
After reading through the code, I understand now. The term "lease" was confusing. Probably something along the lines of "job update timeout" might be a more user friendly flag name. Perhaps it can also be derived from `UpdateJobInterval`, for example `4 * UpdateJobInterval` based on current defaults.

Reply:
Hm, Lease is a common name for this feature, esp in task queues. So I'm not convinced. JobLeaseExpiry or something like that may be an improvement, though.